pkg/platform/runtime/buildscript/buildexpression.go

```diff
@@ -206,6 +207,11 @@ func (s *Script) EqualsBuildExpression(otherJson []byte) bool {
 	return err == nil && string(myJson) == string(otherJson)
 }
 
-func (s *Script) Equals(other *model.BuildExpression) bool {
-	return s.EqualsBuildExpression([]byte(other.String()))
+func (s *Script) Equals(other *buildexpression.BuildExpression) bool {
+	data, err := json.Marshal(other)
+	if err != nil {
+		multilog.Error("Unable to marshal buildexpression to JSON: %v", err)
+		return false
+	}
+	return s.EqualsBuildExpression(data)
```

Comment:
Nit: not technically introduced in this PR but the naming of `Equals()` and `EqualsBuildExpression` feels inverted to me.

Reply:
The naming has always bugged me. I've used even more explicit naming.